components/org.wso2.bfsi.consent.management.common/src/main/java/org/wso2/bfsi/consent/management/common/config/ConsentManagementConfigParser.java

```diff
@@ -0,0 +1,235 @@
+                if (log.isDebugEnabled()) {
+                    log.debug(message.replaceAll("[\r\n]", ""));
+                }
+                throw new FileNotFoundException(message);
+            }
+            builder = new StAXOMBuilder(inStream);
+            OMElement rootElement = builder.getDocumentElement();
+            Stack<String> nameStack = new Stack<>();
+            readChildElements(rootElement, nameStack);
+        } catch (IOException | XMLStreamException | OMException e) {
+            throw new ConsentManagementRuntimeException("Error occurred while building configuration from" +
+                    " bfsi-consent-management.xml", e);
+        } finally {
+            try {
+                if (inStream != null) {
+                    inStream.close();
+                }
+            } catch (IOException e) {
+                log.error("Error closing the input stream for bfsi-consent-management.xml", e);
+            }
+        }
+    }
+
+
+    /**
+     * Method to read text configs from xml when root element is given.
+     *
+     * @param serverConfig XML root element object
+     * @param nameStack    stack of config names
+     */
+    private void readChildElements(OMElement serverConfig, Stack<String> nameStack) {
+
+        for (Iterator childElements = serverConfig.getChildElements(); childElements.hasNext(); ) {
+            OMElement element = (OMElement) childElements.next();
+            nameStack.push(element.getLocalName());
+            if (elementHasText(element)) {
+                String key = getKey(nameStack);
+                Object currentObject = configuration.get(key);
+                String value = replaceSystemProperty(element.getText());
+                if (currentObject == null) {
+                    configuration.put(key, value);
+                } else if (currentObject instanceof ArrayList) {
+                    ArrayList<String> list = (ArrayList) currentObject;
+                    if (!list.contains(value)) {
+                        list.add(value);
+                        configuration.put(key, list);
+                    }
+                } else {
+                    if (!value.equals(currentObject)) {
+                        ArrayList<Object> arrayList = new ArrayList<>(2);
+                        arrayList.add(currentObject);
+                        arrayList.add(value);
+                        configuration.put(key, arrayList);
+                    }
+                }
+            }
+            readChildElements(element, nameStack);
+            nameStack.pop();
+        }
+    }
+
+    /**
+     * Method to check whether config element has text value.
+     *
+     * @param element root element as a object
+     * @return availability of text in the config
+     */
+    private boolean elementHasText(OMElement element) {
+
+        String text = element.getText();
+        return text != null && text.trim().length() != 0;
```

Comment:
can use StringUtils.isEmpty()

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/1/commits/aa7e22c798e360153510f1565d76a7f7c81a4965